test/powershell/engine/Job/Jobs.Tests.ps1

```diff
@@ -1,23 +1,275 @@
+            param($parameters)
+            $job = Get-Job @parameters
+            ValidateJobInfo -job $job -state 'Completed' -hasMoreData $true -command ' 1 + 1 '
+        }
+
+        It 'Can get-job with <property>' -TestCases $getJobSwitches {
+            param($parameters)
+            $job = Get-Job @parameters
+            ValidateJobInfo -job $job -state 'Completed' -hasMoreData $true -Name 'StartedJob'
+        }
+
+        It 'Can get-job with <property>' -TestCases $getJobChildJobs {
+            param($parameters)
+            $jobs = Get-Job @parameters
+            $jobs.Count | Should Be 2
+            ValidateJobInfo -job $jobs[0] -state 'Completed' -hasMoreData $true -Name 'StartedJob'
+            ValidateJobInfo -job $jobs[1] -state 'Completed' -hasMoreData $true
+        }
+    }
+
+
+    Context 'Remove-Job tests' {
+        # The test pattern used here is different from other tests since there is a scoping issue in Pester.
+        # If BeforeEach is used then $removeJobTestCases does not bind when the It is called.
+        # This implementation works around the problem by using a BeforeAll and creating a job inside the It.
+        BeforeAll {
+            $removeJobTestCases = @(
+                @{ property = 'Name'}
+                @{ property = 'Id'}
+                @{ property = 'InstanceId'}
+                @{ property = 'State'}
+            )
+        }
+
+        It 'Can remove-job with <property>' -TestCases $removeJobTestCases {
+            param($property)
+            $jobToRemove = Start-Job -ScriptBlock { 1 + 1 } -Name 'JobToRemove' | Wait-Job
+            $splat = @{ $property = $jobToRemove.$property }
+            Remove-Job @splat
+            Get-Job $jobToRemove -ErrorAction SilentlyContinue | Should BeNullOrEmpty
+        }
+    }
+
+    Context 'Stop-Job tests' {
+        # The test pattern used here is different from other tests since there is a scoping issue in Pester.
+        # If BeforeEach is used then $stopJobTestCases does not bind when the It is called.
+        # This implementation works around the problem by using a BeforeAll and creating a job inside the It.
+        BeforeAll {
+            $stopJobTestCases = @(
+                @{ property = 'Name'}
+                @{ property = 'Id'}
+                @{ property = 'InstanceId'}
+                @{ property = 'State'}
+            )
+        }
+
+        It 'Can stop-job with <property>' -TestCases $stopJobTestCases {
+            param($property)
+            # '-Seconds 100' is chosen to be substantially large, so that the job is in running state when Stop-Job is called.
+            $jobToStop = Start-Job -ScriptBlock { Start-Sleep -Seconds 100 } -Name 'JobToStop'
+            $splat = @{ $property = $jobToStop.$property }
+            Stop-Job @splat
+            ValidateJobInfo -job $jobToStop -state 'Stopped' -hasMoreData $false
+            Remove-Job $jobToStop -Force -ErrorAction SilentlyContinue
```

Comment:
Clean up should be done in an AfterEach. Remove-Job won't execute if the ValidateJobInfo test fails.

Reply:
Fixed.